infra/gravity/node_commands.go

```diff
@@ -587,6 +604,66 @@ func (g *gravity) RunInPlanet(ctx context.Context, cmd string, args ...string) (
 	return out, nil
 }
 
+// IsLeader returns true if node is leader
+func (g *gravity) IsLeader(ctx context.Context) bool {
+	status, err := g.Status(ctx)
+	if err != nil {
+		return false
+	}
+	etcdLeaderKey := fmt.Sprintf("/planet/cluster/%s/master", status.Cluster.Cluster)
+
+	// TODO: use planet cmd once planet cli is updated s.t. `leader view` does
+	// not require args.
+	// leaderIP, err := g.RunInPlanet(ctx, "planet", "leader", "view")
+	leaderIP, err := g.RunInPlanet(ctx, "etcdctl", "get", etcdLeaderKey)
```

Comment:
Why not just use "planet leader view" with the same etcd key flag?

Reply:
Wasn't sure if the command worked properly. Was getting an error when I tested it on a cluster I had running locally. I'll retest it and replace it with proper planet command.